src/pages/todo/components/TodoListItem.js

```diff
@@ -1,34 +1,89 @@
-import React from 'react';
+import React from 'react'
 import { Link } from 'react-router'
+import { UPDATE_TODO } from '../actions'
 
 class TodoListItem extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {editing: false}
+    this.state = {
+      form: { 
+        title: this.props.todo.title
+      },
+      editing: false
+    }
   }
   
   componentDidUpdate() {
     this.refs.textbox.focus()
   }
+
   _showInput() {
     this.setState({editing: true})
   }
 
-  _hideInput() {
+  _updateTodo(event) {
+    const { todo } = this.props
+    let data
+    switch (event.target.type) {
+      case 'checkbox':
+        data = {
+          title: todo.title,
+          marked: (todo.marked == 1) ? 0 : 1
```

Comment:
`marked: (todo.marked == 1) ? 0 : 1`, is this correct? I think it needs checkbox state to derived the `marked` of the `data`.